xen-bugtool

```diff
@@ -1516,6 +1520,38 @@ def multipathd_topology(cap):
 
     return stdout
 
+def snmp_xs_conf(cap):
+    sed_snmp_xs_conf = ['sed',
+    '-e', 's#\(community *= *\).*#\\1********#',
+    '-e', 's#\(authentication_key *= *\).*#\\1********#',
+    '-e', 's#\(privacy_key *= *\).*#\\1********#',
+    '/etc/snmp/snmp.xs.conf']
+    pipe = Popen(sed_snmp_xs_conf, bufsize=1, stdin=PIPE, stdout=PIPE, stderr=dev_null)
+    stdout, _ = pipe.communicate()
+    return stdout
```

Comment:
I'm sorry, but so far, but I see no justifification for calling out to sed when you have `re`.

It's like calling a old 1970-ties taxi from across the country to pick you up for a 5 minute ride when you sit in a sports car with it's motor already running and ready to hit the pedal.

```suggestion
def filter_snmp_xs_conf(_):
    """Filter /etc/snmp/snmp.xs.conf with keys and community removed"""
    with open("/etc/snmp/snmp.xs.conf", "r") as file:
        return re.sub(r"((community|\w_key)\s*=\s*)\S+", r"\1REMOVED", file.read())
```

Fixes:
- Prefix the function name with either `dump_` or in this case more descriptive `filter_` like other dump and filter functions added by `func_output()`.
- Don't trigger `pylint` warning `W0613: Unused argument 'cap' (unused-argument)`
- Add missing function docstring
- Don't call `fork()`, `exceve()`, no dynamic linking load of `sed` and no pipe (less load)
- Fix replacing the secret with the word `REMOVED` as already established by numerous bugtool filters.
- Fix removing the secrets also when tabs (or tabs and spaces) are around the equals sign `=`

If the file is can't be found, the exception is recorded in the output file content, if the output file shall be empty as it is now in the case of such case, just add a test for checking if the file exists before trying to open it.

Reply:
Updated.